include/network/Network.h

```diff
@@ -150,21 +166,46 @@ template <typename T> class Network : public NetworkBase<T> {
             throw std::invalid_argument("Padded weights width cannot be less than the unpadded.");
 
         /// we need to copy everything here since we do not want to have an implicit copy of 'this'
+
         const int padded_input_width = get_input_width();
         const int network_width = get_network_width();
         const int padded_output_width = get_output_width();
+        const int input_width_divisibly_by = 16;
+
+        // to replicate the behaviour of bias (we only do W*x, not W*x+bias), the input has (unpadded_input_width +
+        // one_padded_input_width + zero_padded_input_width) elements, where
+        // zero_padded_input_width = padded_input_width - (unpadded_input_width + one_padded_input_width)
+        // This is following tiny-cuda-nn's behaviour, where the unpadded_input_width is padded up to the next 16th
+        // element with ones. We thus also pad the unpadded_input_width with ones to the next 16th element and fill the
+        // rest with zeros. The input weight matrix in SwiftNet will be of dimension padded_input_width x network_width.
+        // However, only the frist one_padded_input_width rows have values, the rest will have zeros as the input vector
+        // will only have (unpadded_input_width) true elements, then (one_padded_input_width-unpadded_input_width) ones
+        // that serve as bias term. The rest will be zero.
+        int one_padded_input_width = 0;
+        if (!use_bias_) {
+            one_padded_input_width = unpadded_input_width; // we don't pad with ones at all, if not using bias
+        } else if (unpadded_input_width % input_width_divisibly_by == 0) {
+            one_padded_input_width = padded_input_width;
+        } else {
+            one_padded_input_width = (unpadded_input_width / input_width_divisibly_by + 1) *
```

Comment:
I think we have (or should have) a common function which does this calculation

Reply:
https://github.com/Yunaik/tiny-dpcpp-nn/blob/e0a82fdae548c95dafb6f92503103067ecef3765/include/common/common.h#L54